Mathlib/Order/Heyting/Basic.lean

```diff
@@ -1118,17 +1100,15 @@ theorem toDual_sdiff (a b : α) : toDual (a \ b) = toDual b ⇨ toDual a :=
   rfl
 #align to_dual_sdiff toDual_sdiff
 
-instance Prod.coheytingAlgebra [CoheytingAlgebra β] : CoheytingAlgebra (α × β) :=
-  { Prod.lattice α β, Prod.boundedOrder α β, Prod.sdiff α β, Prod.hnot α β with
```

Comment:
Can you split this `where` vs `{}` cleanup to another PR? There seems to be quite a lot of it (which is good), but it seems orthogonal to adjusting the hierarchy.

Reply:
Seems time-consuming to do so because I'm also changing the parents order simultaneously.

Reply:
That instance can be left unchanged, since it doesn't touch `HeytingAlgebra` nor `Frame`

Reply:
Nevermind, a change is required for `OrderDual.instCompleteBooleanAlgebra` to unify the `himp` and `compl` from both `BooleanAlgebra` and `CompleteDistribLattice`

Reply:
I've decided to overdo things a bit to make sure the different structures (`OrderDual`, `Prod`, `Pi` mostly) align in the way their instances are defined

Reply:
I'll make the split

Reply:
#10588. I can deal with the merge conflicts here once that lands.

Reply:
I guess that didn't help with shrinking the diff all that much, though it's certainly more readavle now. The instance renaming looks to be the cause of most of the size here

Reply:
I don't have much more time to spend on this PR for the time being, but I'm happy for you to split off the renaming if you want to.